src/workerd/api/sql.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/jsg/jsg.h>
+#include <workerd/jsg/string.h>
+#include "basics.h"
+
+namespace workerd::api {
+
+typedef kj::OneOf<kj::Array<const byte>, kj::String, double> ValueBind;
+class SqlDatabase;
+
+class SqlResult final: public jsg::Object {
+public:
+  SqlResult(SqliteDatabase::Query& query);
+
+  kj::ArrayPtr<jsg::Dict<SqliteDatabase::Query::ValueOwned>> getResults(jsg::Lock& js);
+
+  JSG_RESOURCE_TYPE(SqlResult, CompatibilityFlags::Reader flags) {
+    JSG_READONLY_PROTOTYPE_PROPERTY(results, getResults);
+  }
+
+private:
+  kj::Vector<jsg::Dict<SqliteDatabase::Query::ValueOwned>> results;
+};
+
+class SqlPreparedStatement final: public jsg::Object {
+public:
+  SqlPreparedStatement(jsg::Ref<SqlDatabase>&& sqlDb, SqliteDatabase::Statement&& statement);
+
+  jsg::Ref<SqlResult> run(kj::Maybe<kj::Array<ValueBind>> bindValuesOptional);
+
+  JSG_RESOURCE_TYPE(SqlPreparedStatement, CompatibilityFlags::Reader flags) {
+    JSG_METHOD(run);
+  }
+
+  void visitForGc(jsg::GcVisitor& visitor) {
```

Comment:
Nitpick: For subclasses of `jsg::Object`, `visitForGc()` should be private. (JSG is still able to find it, via macro magic.)

Reply:
Nice will make it private!